pkg/cache/clusterqueue.go

```diff
@@ -481,14 +481,22 @@ func (c *clusterQueue) addWorkload(log logr.Logger, w *kueue.Workload) error {
 	if _, exist := c.Workloads[k]; exist {
 		return errors.New("workload already exists in ClusterQueue")
 	}
+	c.addOrUpdateWorkload(log, w)
+	return nil
+}
+
+func (c *clusterQueue) addOrUpdateWorkload(log logr.Logger, w *kueue.Workload) {
```

Comment:
This function was introduced based on the main, I put in a dedicated commit https://github.com/kubernetes-sigs/kueue/pull/5334/commits/06fa66a828856af8dc5ef55f3c34b2920d29ce44